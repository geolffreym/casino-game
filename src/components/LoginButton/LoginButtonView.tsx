import React, { FC, ReactElement } from 'react'
import RoundButton from '@components/RoundButton'
import { Button, Transitions } from '@material-ui/core'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import TextField from '@material-ui/core/TextField'
import DialogActions from '@material-ui/core/DialogActions'
import Slide from '@material-ui/core/Slide'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
const Transition = React.forwardRef(function Transition (
  props: Transitions & { children?: React.ReactElement<any, any> },
  ref: React.Ref<unknown>
) {
  return <Slide direction='up' ref={ref} {...props} />
})

interface LoginButtonViewProps {
  open: boolean
  handleOpen: () => void
  handleClose: () => void
  handleLogin: () => void
  handleChangeEmail: (e: any) => void
  handleChangePassword: (e: any) => void
}

const LoginButtonView: FC<LoginButtonViewProps> = ({
  open, handleOpen, handleClose, handleLogin,
  handleChangeEmail, handleChangePassword
}): ReactElement => {
  return (
    <>
      <RoundButton variant='contained' color='primary' onClick={handleOpen}>
        Login
      </RoundButton>
      <Dialog
        open={open}
        TransitionComponent={Transition}
        keepMounted
        onClose={handleClose}
        maxWidth='lg'
        aria-labelledby='alert-dialog-slide-title'
        aria-describedby='alert-dialog-slide-description'
      >
        <DialogTitle id='form-dialog-title'>LOGIN</DialogTitle>
        <DialogContent>
          <TextField
            autoFocus
            margin='dense'
            id='email'
            label='Email Address'
            type='email'
            onChange={handleChangeEmail}
            fullWidth
          />
          <TextField
            autoFocus
            margin='dense'
            id='password'
            label='Password'
            type='password'
            onChange={handleChangePassword}
            fullWidth
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color='primary'>
            Cancel
          </Button>
          <Button onClick={handleLogin} variant='contained' color='primary'>
            Login
          </Button>
        </DialogActions>
      </Dialog>
    </>
  )
}

export default LoginButtonView
