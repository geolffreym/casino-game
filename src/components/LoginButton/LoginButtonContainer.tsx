import LoginButtonView from '@components/LoginButton/LoginButtonView'
import React, { FC } from 'react'
import { connect } from 'react-redux'
import { login, User } from '@state/reducers/userReducers'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
const LoginButtonContainer: FC<{ login: (user: User) => void }> = ({ login }): JSX.Element => {
  const [open, setOpen] = React.useState(false)
  const [email, setEmail] = React.useState('')
  const [password, setPassword] = React.useState('')

  const handleOpen = () => {
    setOpen(true)
  }

  const handleClose = () => {
    setOpen(false)
  }

  const handleLogin = () => {
    login({ email, password })
  }

  const handleChangeEmail = (e: any) => {
    setEmail(e.target.value)
  }

  const handleChangePassword = (e: any) => {
    setPassword(e.target.value)
  }

  return (
    <LoginButtonView {...{ open, handleChangePassword, handleChangeEmail, handleClose, handleLogin, handleOpen }} />
  )
}

export const LoginButton = connect(
  null,
  { login }
)(LoginButtonContainer)
