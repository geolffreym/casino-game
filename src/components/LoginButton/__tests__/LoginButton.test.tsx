import LoginButton from '@components/LoginButton/index'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import { createMemoryHistory } from 'history'
import createStore from '@state/store'
import { Provider } from 'react-redux'

/* eslint-disable no-undef */
/* eslint-disable @typescript-eslint/consistent-type-assertions */
describe('<LoginButton />', () => {
  it('should render', () => {
    const history = createMemoryHistory()
    const store = createStore(history)
    const component = shallow(
      <Provider store={store}>
        <LoginButton />;
      </Provider>
    )
    expect(component).toMatchSnapshot()
  })
})
