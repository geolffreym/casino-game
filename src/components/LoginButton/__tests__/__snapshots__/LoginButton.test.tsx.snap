// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginButton /> should render 1`] = `ShallowWrapper {}`;
